### DISCUSSION TITLE
Invalid Python interpreter when using "maturin develop"

### DISCUSSION THREAD

**Discussion by nednoodlehead** [PyO3/maturin#1121] [Q&A]
Full error:
(venv) F:\Projects\Python Projects\punge\rust>maturin develop     
🔗 Found pyo3 bindings
💥 maturin failed
  Caused by: The given list of python interpreters is invalid
  Caused by: Python interpreter `F:\Files at random\theoretical python\punge\venv\python.exe` doesn't exist

- The path that it is trying to point to is where I used to have my python project.
- As outlined on the [readme](https://github.com/PyO3/maturin#pyo3-and-rust-cpython), I have set "PYO3_PYTHON" and "PYTHON_SYS_EXECUTEABLE" (system and user environmental variable) to a valid python intrepreter
- The path that maturin wants to use is not listed when viewing the environment variables from windows:
![image](https://user-images.githubusercontent.com/102321690/191872710-4fb48933-e901-4f49-ad01-e72e04e7a994.png)
-But does show up in python when using "os.environ["PATH"]".
![image](https://user-images.githubusercontent.com/102321690/191872801-cc7cf8c8-cfe3-42e5-98f9-cef297775aaf.png)
- Running dir env:Path in PowerShell does not show the invalid Python interpreter path for the 'Path' variable
- This is what my User and Machine variables look like:
![image](https://user-images.githubusercontent.com/102321690/191876582-d10b5515-40fa-4e2c-ac5b-65f0e6766268.png)
![image](https://user-images.githubusercontent.com/102321690/191876612-2931c1b9-f1e2-43c7-8388-23edbcfc6441.png)
- The "C:\Users\Spencer\AppData\Local\Programs\Python\Python310\python.exe" or "F:\Projects\Python Projects\punge\venv\Scripts\" path are both valid paths.

Any help would be appreciated

**Comment by messense** [comment]
Sorry I don't have Windows environment, but I think `PYO3_PYTHON` should point to `python.exe` path rather it's directory.

**Reply by nednoodlehead** [reply]
I had originally tried having it point to the exe. Did not work. It seems to not want to use either variable.

**Comment by Skelectric** [comment]
I'm having similar issues, also on windows. 

I have a python interpreter that Maturin is missing - D:\Users\\[my name]\miniconda3\envs\dev\python.exe

Same as you, I added environment variables PYO3_PYTHON and PYTHON_SYS_EXECUTABLE for both user and system, which did not help. 

Running `maturin develop`:

```
⚠️  Warning: Failed to determine python platform
🔗 Found pyo3 bindings
💥 maturin failed
  Caused by: The given list of python interpreters is invalid
  Caused by: Python interpreter `D:\Users\[my name]\miniconda3\envs\dev\Scripts\python.exe` doesn't exist
```

Why is it looking in scripts? 😔

Running `maturin list-python`:

```
🐍 1 python interpreter found:
 - CPython 3.9 at D:\Users\Phil\miniconda3\python.exe
```

It's only finding the interpreter in the base environment, which is the opposite of what it's supposed to be doing according to this blurb from https://github.com/PyO3/maturin#pyo3-and-rust-cpython :

_On windows all versions from the python launcher (which is installed by default by the python.org installer) and all conda environments **except base** are used._

I was hoping that the `maturin list-python --target` option would help here but I can't get it to work either - which I brought up here: https://github.com/PyO3/maturin/pull/957

**Reply by nednoodlehead** [reply]
I actually fixed my issue today, which was embarrassingly easy; pip reinstall maturin.

I'm not sure if it would fix your issue since my issue was more caused by moving folders around, and having that path point to a non-existent python.exe. I'm not sure if you did something similar, moving files / folders around, but one thing is consistent:

The environmental variables do not seem to matter. Maturin does not read them.
